### PR TITLE
Added -tiff option, update ICDeviceDelegate implementation

### PR DESCRIPTION
**Pull request by ks3**
These changes add support for scanning in TIFF format. I also had to add an additional function to the ScannerController class that is now required for ICDeviceDelegate implementation.

### CODE REVIEW

File: scanline/ScanlineAppController.swift [hunk 1]
```diff
@@ -276,7 +284,7 @@ class ScanlineOutputProcessor {
     }
     
     func process() -> Bool {
-        if configuration.config[ScanlineConfigOptionJPEG] != nil {
+        if configuration.config[ScanlineConfigOptionJPEG] != nil || configuration.config[ScanlineConfigOptionTIFF] != nil {
```
Comment (klep) [inline scanline/ScanlineAppController.swift]: Just to make this a little more clear, and perhaps future proof a bit
```suggestion
        let wantsPDF = configuration.config[ScanlineConfigOptionJPEG] == nil && configuration.config[ScanlineConfigOptionTIFF] == nil
        if !wantsPDF {
```
(just to make things a little more clear and perhaps future proof)

**Comment by klep** [comment]
Thanks for the PR @ks3 !

File: scanline/ScanlineAppController.swift (hunk 1 above)
Comment (ks3) [reply scanline/ScanlineAppController.swift]: I definitely agree that ScanConfiguration could benefit from some improvements. One of the things I tested for the changes was how it reacted when both -tiff and -jpeg were given, as there wasn't a built-in way to say only allow one value out of several.

I think the wantsPDF change makes sense. I just pushed it up to my branch.

File: scanline/ScanlineAppController.swift [hunk 2]
```diff
@@ -126,7 +126,10 @@ class ScannerController: NSObject, ICScannerDeviceDelegate {
         logger.verbose("didEncounterError: \(error?.localizedDescription ?? "[no error]")")
         delegate?.scannerControllerDidFail(self)
     }
-    
+    func device(_ device: ICDevice, didCloseSessionWithError error: Error?) {
```
Comment (klep) [inline scanline/ScanlineAppController.swift]: @ks3 When I compile, I get a warning that `Error` isn't supposed to be optional. Are you seeing that too? Also, mind inserting a blank line between these delegate methods?

Comment (ks3) [reply scanline/ScanlineAppController.swift]: Well shoot. The main doc page (https://developer.apple.com/documentation/imagecapturecore/icdevicedelegate) lists it as an optional, but looks like that's only on iOS. I should be able to get an updated commit pushed this evening.

**Pull request merged**